adum7701_fmc/clkgen_core.c

```diff
@@ -425,5 +425,7 @@ int32_t clkgen_setup(clkgen_device **clkgen_dev,
 		status--;
 	}
 
+	free (adc_clkgen);

```

Comment:
I think in this case you could probably change the type of the `adc_clkgen` variable;
to `clkgen_device adc_clkgen;`

that way it's automatically allocated on & free'd from the stack

Reply:
this can be done here, also because you are allocating a fixed size memory block, so malloc() doesn't add much value